### PR TITLE
Rename compile task for compatibility with :compilers option

### PR DESCRIPTION
**Pull request by michallepicki**
Also: deprecate MixGleam.add_aliases

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -36,7 +36,8 @@ compiler and Gleam dependencies:
       app: @app,
       # ...
       archives: [mix_gleam: "~> 0.4.0"],
-      aliases: MixGleam.add_aliases(aliases()),
+      compilers: [:gleam | Mix.compilers()],
```
Comment (tynanbe) [inline README.md:39]: We should put a note here about how to integrate with Phoenix, which looks to use `aliases()` for its own aliases. See #16.

File: lib/mix_gleam.ex [hunk 2]
```diff
@@ -19,10 +19,11 @@ defmodule MixGleam do
     end)
   end
 
+  @deprecated "Please prepend :gleam to :compilers and add :\"deps.get\" to :aliases following mix_gleam README.md instead"
```
Comment (tynanbe) [inline lib/mix_gleam.ex:22]: I'm uncertain about deprecating this functionality.
I guess it depends on whether we're likely to have any other aliases in the future; I agree it isn't much use with only one alias, and it's nice to have a more explicit config in that case.

If we do deprecate this function, the https://github.com/gleam-lang/mix_gleam/blob/main/lib/mix_gleam/aliases.ex module should probably also be deprecated.

@lpil?

**Review by tynanbe — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (michallepicki) [reply README.md:39]: So for Phoenix the function is a private helper in `mix.exs` defined at the bottom on that file. What I usually do and see in other people's projects is that new aliases are being added to that helper function. Seeing the instructions from this PR I personally would know what to do, just by following the `aliases()` function to its definition, and I would paste the alias there. Maybe something like a comment with `# or add this alias to your aliases() function` would suffice?

File: lib/mix_gleam.ex (hunk 2 above)
Comment (michallepicki) [reply lib/mix_gleam.ex:22]: My goal with this PR were:
- Make it easier for people to see what the plugin is doing and how it works. I think the `add_aliases` function obscures some of it and directly adding an alias is easier to understand.
- Bring the `mix_gleam` project a bit closer to not needing to be an archive. I think if a function from this project is being called in `mix.exs` then it needs to be an archive. Next step would be to make Mix understand gleam sub-dependencies fetching (I think?)
- Have one way to do things

That's why I think it's beneficial to deprecate this function and have users change their setup. I would consider the `MixGleam.Aliases` module to be internal and not public interface, so I don't think it needs to be deprecated

**Review by tynanbe — APPROVED** [review]

File: lib/mix_gleam.ex (hunk 2 above)
Comment (tynanbe) [reply lib/mix_gleam.ex:22]: Sounds reasonable to me.

**Pull request merged**